server/src/main/java/org/elasticsearch/index/SlowLogLevel.java

```diff
@@ -21,8 +21,23 @@
 import java.util.Locale;
 
 public enum SlowLogLevel {
-    WARN, TRACE, INFO, DEBUG;
+    WARN(0), //most specific - little logging
```

Comment:
I find it somewhat counterintuitive that the most specific logging level has the lowest "value" here. Perhaps change the meaning from "precedence" to "specificity" and then the values can be:

```
WARN(3)
INFO(2)
DEBUG(1)
TRACE(0)
```

and then the comparison can be `<=` instead of `>=` which I find more intuitive. That is:

```
SlowLogLevel.TRACE.isLevelEnabledFor(SlowLogLevel.WARN)
```

amounts to a comparison of

```
0 <= 3
```

which is true, and makes sense because `TRACE` is *less* specific than `WARN`.

Reply:
I tried to follow the ordering they use in `org.apache.logging.log4j.spi.StandardLevel` where the lower the number the less logs it will allow. OFF being 0 and trace being highest number.

But I also find your reasoning good. The higher the number the more important the log is.

I am on the fence now :)

Reply:
as per our discussion I think we should use your ordering and wording. The more specific the log level the higher the `specificity` value and less logs.